docs/development/patches.md

```diff
@@ -0,0 +1,84 @@
+
+## Patch system
+
+If you find yourself in the unfortunate position of having to make a change which can only be made through patching an upstream project, you'll need to know how to manage patches in Electron.
+
+All patches to upstream projects in Electron are contained in the `patches/` directory. Each subdirectory of `patches/` contains several patch files, along with a `.patches` file which lists the order in which the patches should be applied. Think of these files as making up a series of git commits that are applied on top of the upstream project after we check it out.
+
+```
+patches
+├── config.json   <-- this describes which patchset directory is applied to what project
+├── chromium
+│   ├── .patches
+│   ├── accelerator.patch
+│   ├── add_contentgpuclient_precreatemessageloop_callback.patch
+│   ⋮
+├── node
+│   ├── .patches
+│   ├── add_openssl_is_boringssl_guard_to_oaep_hash_check.patch
+│   ├── build_add_gn_build_files.patch
+│   ⋮
+⋮
+```
+
+To help manage these patch sets, we provide two tools: `git-import-patches` and `git-export-patches`. `git-import-patches` imports a set of patch files into a git repository by applying each patch in the correct order and creating a commit for each one. `git-export-patches` does the reverse; it exports a series of git commits in a repository into a set of files in a directory and an accompanying `.patches` file.
+
+> Side note: the reason we use a `.patches` file to maintain the order of applied patches, rather than prepending a number like `001-` to each file, is because it reduces conflicts related to patch ordering. It prevents the situation where two PRs both add a patch at the end of the series with the same numbering and end up both getting merged resulting in a duplicate identifier, and it also reduces churn when a patch is added or deleted in the middle of the series.
+
+### Usage
+
+#### Adding a new patch
+```
+$ cd src/third_party/electron_node
+$ vim some/code/file.cc
+$ git commit
+$ ../../electron/script/git-export-patches -o ../../electron/patches/node
+```
+
+Re-exporting patches will sometimes cause shasums in unrelated patches to change. This is generally harmless and can be ignored (but go ahead and add those changes to your PR, it'll stop them from showing up for other people).
+
+#### Editing an existing patch
+```
+$ cd src/v8
+$ vim some/code/file.cc
+$ git log
+# Find the commit sha of the patch you want to edit.
+$ git commit --fixup [COMMIT_SHA]
+$ git rebase --autosquash -i [COMMIT_SHA]^
```

Comment:
This is a much easier way of editing a patch than the way I was doing it.  🤦‍♂

Reply:
Glad to be of service!